reports.py

```diff
@@ -550,6 +550,12 @@ def parser_plot_coverage_common(parser=argparse.ArgumentParser()):    # parser n
         help="The max coverage depth (default: %(default)s)"
     )
     parser.add_argument('-l', dest="read_length_threshold", default=None, type=int, help="Read length threshold")
+    parser.add_argument(
+        '--binLargePlots',
+        dest="bin_large_plots",
+        action="store_true",
```

Comment:
Do you think we should turn on binned plots as the default option for genomes beyond a certain length (the axis `xlim`?)? It would be nice for users to simply get a plot regardless of genome size, and your y-axis labeling code makes it clear whether a given plot has been binned or not. If we do that we'd probably want the ability to override and turn off auto-binning (or set the bin size manually and auto-scale the width?).

Reply:
Beyond some point, plots without binning are useless. But there is a middle ground where you could bin or not bin. We could have binning turned on by default with an override option, and bin a bit less aggressively (something like two to 20 bins per pixel, rather than one bin per pixel). If a plot doesn't really need it, then it won't be binned. As things currently are, GB virus C gets 11-bp bins when binning is turned out, but the non-binned plot is quite readable and reasonably sized, which makes me think I was a bit too aggressive.

Setting the bin size manually—I would not make this the default, but I think that would make an excellent option—I can see people wanting to be able to do it. Auto-scaling the width would be a nice way to make sure that all coverage plots produced are actually readable, but I don't think it is absolutely necessary.